### PR TITLE
Pull request for no-g_pull

### PR DESCRIPTION
**Pull request by sileht**
## refactor: start removing usage of g_pull

This is a first step for removing pygithub.

MergifyPull will act as proxy to pygithub.PullRequest attributes

Then we will have almost all pygithub call located here.

### CODE REVIEW

File: mergify_engine/mergify_pull.py [hunk 1]
```diff
@@ -342,3 +337,85 @@ def __str__(self):
...
+    @property
+    def merged(self):
+        return self.g_pull.merged
+
+    @property
+    def merged_by(self):
+        return self.g_pull.merged_by
+
+    @property
+    def stat_commits(self):
```
Comment (jd) [inline mergify_engine/mergify_pull.py]: why `stat_commits`?
Is this the number of commits? Should be `num_commits` then?

File: mergify_engine/mergify_pull.py [hunk 2]
```diff
@@ -342,3 +337,85 @@ def __str__(self):
...
+    @property
+    def base_repo_name(self):
+        return self.g_pull.base.repo.name
+
+    @property
+    def head_repo_name(self):
+        return self.g_pull.head.repo.name
+
+    @property
+    def base_repo_owner(self):
```
Comment (jd) [inline mergify_engine/mergify_pull.py]: `base_repo_owner_login`?

File: mergify_engine/mergify_pull.py [hunk 3]
```diff
@@ -342,3 +337,85 @@ def __str__(self):
...
+        return self.g_pull.base.repo.owner.login
+
+    @property
+    def head_repo_owner(self):
```
Comment (jd) [inline mergify_engine/mergify_pull.py]: `head_repo_owner_login`?

File: mergify_engine/mergify_pull.py [hunk 4]
```diff
@@ -342,3 +337,85 @@ def __str__(self):
...
+    @property
+    def user(self):
+        return self.g_pull.user.login
+
+    @property
+    def state(self):
+        return self.g_pull.state
+
+    @property
+    def come_from_same_repo(self):
```
Comment (jd) [inline mergify_engine/mergify_pull.py]: I'd rather do `from_fork` it's simpler as a name, just invert the result

File: mergify_engine/mergify_pull.py (hunk 1 above)
Comment (sileht) [reply mergify_engine/mergify_pull.py]: This is not even needed.

**Pull request merged**